### PR TITLE
tests: subsys/pm: move to new ztest API

### PR DESCRIPTION
**Pull request by yinfangw**
Move test subsys/pm to use new ztest API.

Signed-off-by: Yinfang Wang yinfang.wang@intel.com

### CODE REVIEW

**Review by gmarull — APPROVED** [review]

File: tests/subsys/pm/device_runtime_api/src/main.c [hunk 1]
```diff
@@ -236,15 +236,12 @@ static void test_unsupported(void)
 	zassert_equal(pm_device_runtime_put(dev), -ENOTSUP, "");
 }
 
-void test_main(void)
+void *device_runtime_api_setup(void)
```
Comment (ceolin) [inline tests/subsys/pm/device_runtime_api/src/main.c:239]: why this function is returning a pointer ? Shouldn't it be a `void` function ... it always return NULL.

Comment (yinfangw) [reply tests/subsys/pm/device_runtime_api/src/main.c:239]: these lines can answer you question: 

ZTEST_SUITE calls the struct ztest_suite_node, then setup defintion, please check. 

- https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/testsuite/ztest/include/zephyr/ztest_test_new.h#L202-L213
- https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/testsuite/ztest/include/zephyr/ztest_test_new.h#L161-L182
- https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/testsuite/ztest/include/zephyr/ztest_test_new.h#L125

Comment (yinfangw) [reply tests/subsys/pm/device_runtime_api/src/main.c:239]: @ceolin please see the comment, and review the PR again.

**Review by enjiamai — APPROVED** [review]

File: tests/subsys/pm/device_runtime_api/src/main.c (hunk 1 above)
Comment (ceolin) [reply tests/subsys/pm/device_runtime_api/src/main.c:239]: yep, thanks for clarifying it.

**Review by ceolin — APPROVED** [review]

**Pull request merged**